broker/core/utils/src/HttpClient.js

```diff
@@ -50,10 +51,27 @@ class HttpClient {
 
     // If options has rejectUnauthorized: false,
     // create an https agent with this option and pass to axios.
-    if (_.get(options, 'rejectUnauthorized', true) == false) {
-      const httpsAgent = new https.Agent({
+    const reject_unauthorized = _.get(options, 'rejectUnauthorized', true);
+    const mtlsEnabled = _.get(options, 'mtlsEnabled', false);
+    let httpsAgent;
+    if(mtlsEnabled) {
+      const cert = fs.readFileSync('/etc/certs/cert', 'utf8');
+      const key = fs.readFileSync('/etc/certs/key', 'utf8');
+      if (_.isEmpty(cert) || _.isEmpty(key)) {
+        logger.error('Either certificate or key required for quota app authentication, is missing');
+        throw new InternalServerError('Missing certificate or key required to authenticate quota API call');
+      }
+      httpsAgent = new https.Agent({
+        rejectUnauthorized: reject_unauthorized,
+        cert: cert,
+        key: key
```

Comment:
Hi @Pooja-08, please add timeout option here... probably as enhancement in next pr

Reply:
sure.. will do so